chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/listing/reviewer/edition2015/OldCriteriaWithoutIcsReviewer.java

```diff
@@ -0,0 +1,110 @@
+    private static final String D3_CRITERIA_NUMBER = "170.315 (d)(3)";
+    private static final String D10_CRITERIA_NUMBER = "170.315 (d)(10)";
+    private static final String RULE_EFFECTIVE_DATE = "cures.ruleEffectiveDate";
+
+    private Environment env;
+    private ErrorMessageUtil msgUtil;
+
+    @Autowired
+    public OldCriteriaWithoutIcsReviewer(Environment env, ErrorMessageUtil msgUtil) {
+        this.env = env;
+        this.msgUtil = msgUtil;
+    }
+
+    @Override
+    public void review(CertifiedProductSearchDetails listing) {
+        if (hasICS(listing)) {
+            return;
+        }
+        long b3ChangeDate;
+        long ruleEffectiveDate;
+        try {
+            b3ChangeDate = getPropertyDate(B3_CHANGE_DATE);
+            ruleEffectiveDate = getPropertyDate(RULE_EFFECTIVE_DATE);
+        } catch (Exception e) {
+            LOGGER.error(e.getMessage(), e);
+            return;
+        }
+
+        String error = checkForError(listing, B3_CRITERIA_NUMBER, b3ChangeDate);
+        if (!StringUtils.isEmpty(error)) {
+            listing.getErrorMessages().add(error);
+        }
+        error = checkForError(listing, D2_CRITERIA_NUMBER, ruleEffectiveDate);
+        if (!StringUtils.isEmpty(error)) {
+            listing.getErrorMessages().add(error);
+        }
+        error = checkForError(listing, D3_CRITERIA_NUMBER, ruleEffectiveDate);
+        if (!StringUtils.isEmpty(error)) {
+            listing.getErrorMessages().add(error);
+        }
+        error = checkForError(listing, D10_CRITERIA_NUMBER, ruleEffectiveDate);
+        if (!StringUtils.isEmpty(error)) {
+            listing.getErrorMessages().add(error);
+        }
+    }
+
+    private String checkForError(CertifiedProductSearchDetails listing, String criteria, long relevantDate) {
+        long certificationDate = listing.getCertificationDate();
+        if (hasRelevantCriteria(listing, criteria) && certificationDate >= relevantDate) {
+            return getErrorMessage("listing.criteria.hasOldVersionOfCriteria", criteria,
+                    Util.getDateFormatter().format(relevantDate));
+        }
+        return null;
+    }
+
+    private String getErrorMessage(String messageCode, String criteria, String relevantDate) {
+        return msgUtil.getMessage(messageCode, criteria, relevantDate);
+    }
+
+    private long getPropertyDate(String date) throws ParseException {
+        String dateAsString = env.getProperty(date);
+        SimpleDateFormat sdf = new SimpleDateFormat("MM/dd/yyyy");
+        return sdf.parse(dateAsString).getTime();
+    }
+
+    private boolean hasRelevantCriteria(CertifiedProductSearchDetails listing, String criteria) {
+        return listing.getCertificationResults().stream()
+                .anyMatch(result -> result.getCriterion().getNumber().equals(criteria)
+                        && result.isSuccess()
+                        && !CertificationCriterionService.hasCuresInTitle(result.getCriterion()));
```

Comment:
I guess since the ticket is in final code review you don't have to/shouldn't change this but... I don't like checking against the criterion number and title together. We got bit once by using criterion numbers everywhere and spent a lot of time changing it all to use the IDs. I think you could use the criterion IDs from whatever lookup file we have and check for ID equality rather than number and title.